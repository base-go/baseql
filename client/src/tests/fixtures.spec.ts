import ts from "typescript";
import path from "path";
import { readdirSync } from "fs";

describe("fixtures", () => {
  readdirSync(path.join(__dirname, "fixtures")).forEach(fixture => {
    it(fixture, () => {
      const program = ts.createProgram(
        [path.join(__dirname, "fixtures", fixture)],
        {
          jsx: ts.JsxEmit.React,
          strict: true,
          noEmit: true,
          module: ts.ModuleKind.ES2015,
          target: ts.ScriptTarget.ES2015,
          allowSyntheticDefaultImports: true,
          esModuleInterop: true,
          typeRoots: ["../../node_modules/@types"],
          types: ["react", "lodash.isequal"],
          moduleResolution: ts.ModuleResolutionKind.NodeJs,
        },
      );
      const preEmitDiagnostics = ts
        .getPreEmitDiagnostics(program)
        // Get rid of file field so that diagnostics can match in snapshot testing.
        .map(err => ({ ...err, file: undefined }));

      expect(preEmitDiagnostics).toMatchSnapshot();
    });
  });
});
