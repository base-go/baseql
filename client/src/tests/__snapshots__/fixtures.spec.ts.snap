// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures connection.ts 1`] = `Array []`;

exports[`fixtures hoc.tsx 1`] = `Array []`;

exports[`fixtures hoc_errors.tsx 1`] = `
Array [
  Object {
    "category": 1,
    "code": 2554,
    "file": undefined,
    "length": 83,
    "messageText": "Expected 2-3 arguments, but got 2.",
    "reportsUnnecessary": undefined,
    "start": 485,
  },
  Object {
    "category": 1,
    "code": 2345,
    "file": undefined,
    "length": 12,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type 'QuerySpec<Result, Variables>' is not assignable to parameter of type 'QuerySpec<Result, undefined>'.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type 'Variables' is not assignable to type 'undefined'.",
        "next": undefined,
      },
    },
    "start": 644,
  },
  Object {
    "category": 1,
    "code": 2554,
    "file": undefined,
    "length": 85,
    "messageText": "Expected 2-3 arguments, but got 3.",
    "reportsUnnecessary": undefined,
    "start": 776,
  },
  Object {
    "category": 1,
    "code": 2345,
    "file": undefined,
    "length": 28,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type 'QuerySpec<Result, undefined>' is not assignable to parameter of type 'QuerySpec<Result, object>'.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type 'undefined' is not assignable to type 'object'.",
        "next": undefined,
      },
    },
    "start": 928,
  },
]
`;

exports[`fixtures hoc_without_variables.tsx 1`] = `Array []`;

exports[`fixtures mutation.tsx 1`] = `Array []`;

exports[`fixtures mutation_without_variables.tsx 1`] = `Array []`;

exports[`fixtures query.tsx 1`] = `Array []`;

exports[`fixtures query_with_variables_error.tsx 1`] = `
Array [
  Object {
    "category": 1,
    "code": 2322,
    "file": undefined,
    "length": 5,
    "messageText": Object {
      "category": 1,
      "code": 2322,
      "messageText": "Type '{ children: (data: GraphQLResult<Result>) => null; query: string; }' is not assignable to type 'IntrinsicAttributes & QueryPropsBase<Result, Variables> & { variables: Variables; }'.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type '{ children: (data: GraphQLResult<Result>) => null; query: string; }' is not assignable to type '{ variables: Variables; }'.",
        "next": Object {
          "category": 1,
          "code": 2324,
          "messageText": "Property 'variables' is missing in type '{ children: (data: GraphQLResult<Result>) => null; query: string; }'.",
          "next": undefined,
        },
      },
    },
    "start": 293,
  },
  Object {
    "category": 1,
    "code": 2322,
    "file": undefined,
    "length": 5,
    "messageText": Object {
      "category": 1,
      "code": 2322,
      "messageText": "Type '{ children: (data: GraphQLResult<Result>) => null; query: QuerySpec<Result, Variables>; }' is not assignable to type 'IntrinsicAttributes & QueryPropsBase<Result, Variables> & { variables: Variables; }'.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type '{ children: (data: GraphQLResult<Result>) => null; query: QuerySpec<Result, Variables>; }' is not assignable to type '{ variables: Variables; }'.",
        "next": Object {
          "category": 1,
          "code": 2324,
          "messageText": "Property 'variables' is missing in type '{ children: (data: GraphQLResult<Result>) => null; query: QuerySpec<Result, Variables>; }'.",
          "next": undefined,
        },
      },
    },
    "start": 381,
  },
  Object {
    "category": 1,
    "code": 2326,
    "file": undefined,
    "length": 14,
    "messageText": Object {
      "category": 1,
      "code": 2326,
      "messageText": "Types of property 'variables' are incompatible.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type '{}' is not assignable to type 'Variables'.",
        "next": Object {
          "category": 1,
          "code": 2324,
          "messageText": "Property 'y' is missing in type '{}'.",
          "next": undefined,
        },
      },
    },
    "start": 491,
  },
  Object {
    "category": 1,
    "code": 2326,
    "file": undefined,
    "length": 14,
    "messageText": Object {
      "category": 1,
      "code": 2326,
      "messageText": "Types of property 'variables' are incompatible.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type '{}' is not assignable to type 'Variables'.",
        "next": Object {
          "category": 1,
          "code": 2324,
          "messageText": "Property 'y' is missing in type '{}'.",
          "next": undefined,
        },
      },
    },
    "start": 610,
  },
]
`;

exports[`fixtures query_without_variables.tsx 1`] = `Array []`;

exports[`fixtures query_without_variables_errors.tsx 1`] = `
Array [
  Object {
    "category": 1,
    "code": 2326,
    "file": undefined,
    "length": 14,
    "messageText": Object {
      "category": 1,
      "code": 2326,
      "messageText": "Types of property 'variables' are incompatible.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type '{}' is not assignable to type 'undefined'.",
        "next": undefined,
      },
    },
    "start": 283,
  },
  Object {
    "category": 1,
    "code": 2326,
    "file": undefined,
    "length": 14,
    "messageText": Object {
      "category": 1,
      "code": 2326,
      "messageText": "Types of property 'variables' are incompatible.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type '{}' is not assignable to type 'undefined'.",
        "next": undefined,
      },
    },
    "start": 394,
  },
]
`;
